crypto/src/hash.rs

```diff
@@ -31,16 +31,21 @@ use ursa::blake2::{
     Serialize,
     IntoSchema,
 )]
-pub struct Hash(pub [u8; Self::LENGTH]);
+pub struct Hash([u8; Self::LENGTH]);
 
 impl Hash {
     /// Length of hash
     pub const LENGTH: usize = 32;
 
+    /// Construct zeroed hash
+    pub const fn zeroed() -> Self {
```

Comment:
we should either have this associated method return all 0 hash as it currently is, or remove it and use `Option::None`

Reply:
`Option::None` is more idiomatic. We [need to mark the hash as](https://users.rust-lang.org/t/niche-optimization-nonzero-and-improper-ctypes/41399) `non-zero` so that it doesn't also take up more space.